src/guiapi/src/st7789v.c

```diff
@@ -437,169 +442,201 @@ void st7789v_clip_rect(rect_ui16_t rc) {
+        for (n = ady; n > 0; --n) {
+            st7789v_set_pixel(pt0, clr);
             if ((cy -= cx) <= 0) {
-                pt.x += sx;
-                cy += dy;
+                pt0.x += sx;
+                cy += ady;
             }
-            pt.y += sy;
-        }
-    else //dx == dy
-        for (n = dx; n > 0; n--) {
-            st7789v_set_pixel(pt, clr);
-            pt.x += sx;
-            pt.y += sy;
+            pt0.y += sy;
         }
+        return;
+    }
+
+    //adx == ady => diagonal line
+    for (n = adx; n > 0; --n) {
+        st7789v_set_pixel(pt0, clr);
+        pt0.x += sx;
+        pt0.y += sy;
+    }
 }
 
+// Draws a rectangle boundary of defined color
 void st7789v_draw_rect(rect_ui16_t rc, color_t clr) {
-    point_ui16_t pt = { rc.x, rc.y };
+    point_ui16_t pt0 = { rc.x, rc.y };
     point_ui16_t pt1 = { rc.x + rc.w - 1, rc.y };
-    point_ui16_t pt2 = { rc.x + rc.w - 1, rc.y + rc.h - 1 };
-    point_ui16_t pt3 = { rc.x, rc.y + rc.h - 1 };
-    st7789v_draw_line(pt, pt1, clr);
-    st7789v_draw_line(pt1, pt2, clr);
-    st7789v_draw_line(pt2, pt3, clr);
-    st7789v_draw_line(pt3, pt, clr);
-    st7789v_set_pixel(pt2, clr); //hack, drawline is broken
+    point_ui16_t pt2 = { rc.x, rc.y + rc.h - 1 };
+
+    st7789v_fill_rect(rect_ui16(pt0.x, pt0.y, rc.w, 1), clr); // top
+    st7789v_fill_rect(rect_ui16(pt0.x, pt0.y, 1, rc.h), clr); // left
+    st7789v_fill_rect(rect_ui16(pt1.x, pt1.y, 1, rc.h), clr); // right
+    st7789v_fill_rect(rect_ui16(pt2.x, pt2.y, rc.w, 1), clr); // bottom
 }
 
+// Draws a solid rectangle of defined color
 void st7789v_fill_rect(rect_ui16_t rc, color_t clr) {
     rc = rect_intersect_ui16(rc, st7789v_clip);
     if (rect_empty_ui16(rc))
         return;
-    int i;
-    uint16_t clr565 = _COLOR_565(clr);
-    uint32_t size = (uint32_t)rc.w * rc.h * 2;
-    int n = size / sizeof(st7789v_buff);
-    int s = size % sizeof(st7789v_buff);
-    if (n)
-        st7789v_fill_ui16((uint16_t *)st7789v_buff, clr565, sizeof(st7789v_buff) / 2);
-    else
-        st7789v_fill_ui16((uint16_t *)st7789v_buff, clr565, size / 2);
+
+    uint16_t clr565 = _COLOR_TO_565(clr);
+    uint32_t size = (uint32_t)rc.w * rc.h * 2; // area of rectangle
+
+    st7789v_fill_ui16((uint16_t *)st7789v_buff, clr565, MIN(size, sizeof(st7789v_buff) / 2));
     st7789v_clr_cs();
     st7789v_cmd_caset(rc.x, rc.x + rc.w - 1);
     st7789v_cmd_raset(rc.y, rc.y + rc.h - 1);
     st7789v_cmd_ramwr(0, 0);
-    for (i = 0; i < n; i++)
+
+    for (int i = 0; i < size / sizeof(st7789v_buff); i++) // writer buffer by buffer
         st7789v_wr(st7789v_buff, sizeof(st7789v_buff));
-    if (s)
-        st7789v_wr(st7789v_buff, s);
+
+    st7789v_wr(st7789v_buff, size % sizeof(st7789v_buff)); // write the remainder data
     st7789v_set_cs();
 }
 
-void st7789v_draw_char(point_ui16_t pt, char chr, font_t *pf, color_t clr0, color_t clr1) {
+// Draws a single character accorign to selected font
```

Comment:
accorign ... typo ;)
please prefer using /// for doxygen style comments and @return @param for keywords